src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -238,6 +239,34 @@ public final HttpServer forwarded(boolean forwardedEnabled) {
 		}
 	}
 
+	/**
+	 * Specifies whether support for the {@code "HAProxy proxy protocol"}
+	 * for deriving information about the address of the remote peer is enabled.
+	 *
+	 * @param proxyProtocolEnabled if true support for the {@code "HAProxy proxy protocol"}
+	 *                         for deriving information about the address of the remote peer is enabled,
+	 *                         otherwise disabled.
+	 * @return a new {@link HttpServer}
+	 */
+	public final HttpServer proxyProtocol(boolean proxyProtocolEnabled) {
+		if (proxyProtocolEnabled) {
+			return tcpConfiguration(tcpServer ->
+					tcpServer.bootstrap(b -> BootstrapHandlers.updateConfiguration(b,
+							NettyPipeline.ProxyProtocolDecoder,
+							(connectionObserver, channel) -> {
+								channel.pipeline()
+										.addFirst(NettyPipeline.ProxyProtocolDecoder, new HAProxyMessageDecoder());
+								channel.pipeline()
+										.addAfter(NettyPipeline.ProxyProtocolDecoder,
+												NettyPipeline.ProxyProtocolReader, new HAProxyMessageReader());
+							})));
+		}
+		else {
+			return tcpConfiguration(tcpServer ->
+					tcpServer.bootstrap(b -> BootstrapHandlers.removeConfiguration(b, NettyPipeline.ProxyProtocolDecoder)));
+		}
```

Comment:
Remove `NettyPipeline.ProxyProtocolReader` also

Reply:
NettyPipeline.ProxyProtocolDecoder used here is just a Configuration name, not channelHander name, so NettyPipeline.ProxyProtocolReader is no need to remove here.

Reply:
oh yes, correct.
Thanks for checking that.